src/main/java/name/valery1707/kaitai/KaitaiMojo.java

```diff
@@ -179,51 +185,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 		project.addCompileSourceRoot(generatedRoot.normalize().toFile().getAbsolutePath());
 	}
 
-	static URL prepareUrl(URL url, String version) throws MojoExecutionException {
-		if (url == null) {
-			try {
-				url = new URL(format(URL_FORMAT, version, version));
-			} catch (MalformedURLException e) {
-				throw new MojoExecutionException("Invalid version: " + version, e);
-			}
-		}
-		return url;
-	}
-
-	static Path prepareCache(File target, MavenSession session, Log log) throws MojoExecutionException {
-		Path cache;
-		if (target == null) {
-			Path repository = new File(session.getLocalRepository().getBasedir()).toPath();
-			cache = repository.resolve(".cache").resolve("kaitai").normalize();
+	private Path detectCacheDir() {
+		if (cacheDir != null) {
+			return cacheDir.toPath();
 		} else {
-			cache = target.toPath();
-		}
-		log.debug(format(
-			"KaiTai distribution: Prepare cache directory: %s"
-			, cache.normalize().toFile().getAbsolutePath()
-		));
-		return mkdirs(cache);
-	}
-
-	private static final Map<Boolean, String> SCRIPT_SUFFIX_REMOVER = Collections.unmodifiableMap(new HashMap<Boolean, String>() {{
-		put(true, "");
-		put(false, ".bat");
-	}});
-
-	static Path downloadKaitai(URL url, Path cacheDir, Log log) throws MojoExecutionException {
-		Path distZip = cacheDir.resolve(FilenameUtils.getName(url.getFile()));
-		download(url, distZip, log);
-		Path dist = unpack(distZip, log);
-		List<Path> bats = scanFiles(dist, new String[]{KAITAI_START_SCRIPT}, new String[0]);
-		if (bats.size() != 1) {
-			throw new MojoExecutionException(format(
-				"Fail to find start script '%s' in Kaitai distribution: %s"
-				, KAITAI_START_SCRIPT
-				, dist.normalize().toFile().getAbsolutePath()
-			));
+			Path repository = new File(session.getLocalRepository().getBasedir()).toPath();
+			return repository.resolve(".cache").resolve("kaitai").normalize();
```

Comment:
\[Checkstyle\] WARNING: 'else' child has incorrect indentation level 24, expected level should be 6\.